validator/cycle/process_tasks.py

```diff
@@ -62,6 +63,15 @@ async def _make_offer(node: Node, request: MinerTaskRequest, config: Config) ->
     )
 
 
+def _calculate_selection_probabilities(nodes: list[Node], random_weight: float = 0.005):
+    incentives = np.array([node.incentive for node in nodes])
+    normalized_incentives = incentives / np.sum(incentives)
+    random_probs = np.random.random(len(nodes))
+    random_probs = random_probs / np.sum(random_probs)
+    selection_probs = (1 - random_weight) * normalized_incentives + random_weight * random_probs
```

Comment:
Why do we have random weight in at all? Why not just clip the incentive weights.

Reply:
murdered